llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -2140,8 +2140,9 @@ InstructionCost RISCVTTIImpl::getMemoryOpCost(unsigned Opcode, Type *Src,
   // Assume memory ops cost scale with the number of vector registers
   // possible accessed by the instruction.  Note that BasicTTI already
   // handles the LT.first term for us.
-  if (LT.second.isVector() && CostKind != TTI::TCK_CodeSize)
-    BaseCost *= TLI->getLMULCost(LT.second);
+  if (TLI->getSubtarget().hasVInstructions())
+    if (LT.second.isVector() && CostKind != TTI::TCK_CodeSize)
```

Comment:
Combine the 2 `if` conditions with `&&` instead of nesting

Reply:
Done, thanks for pointing out.

It seems like there are actually a lot of `ST->hasVInstructions()` checks in the file, so it seems like we are doing the right thing here.